src/util/Logging.cpp

```diff
@@ -111,18 +258,23 @@ Logging::getStringFromLL(el::Level level)
 bool
 Logging::logDebug(std::string const& partition)
 {
+    std::lock_guard<std::recursive_mutex> guard(mLogMutex);
     return mLogLevel <= el::Level::DEBUG;
 }
 
 bool
 Logging::logTrace(std::string const& partition)
 {
+    std::lock_guard<std::recursive_mutex> guard(mLogMutex);
     return mLogLevel <= el::Level::TRACE;
 }
 
 void
 Logging::rotate()
 {
+    std::lock_guard<std::recursive_mutex> guard(mLogMutex);
```

Comment:
Does it mean that when rotating logs it forgets the current log level? I see that `setLogLevel` performs a bunch of stuff that we're not redoing in `init`

Reply:
oh, yeah, oops. fixed.